[FRONT-1461] Navigation update

pmichalowski opened the pull request.

mondoreale reviewed (commented):
Looks good!

Few notes

1. Missing border-radius on hover for edge items:
![Screenshot 2023-06-26 at 3 47 21 PM](https://github.com/streamr-dev/core-frontend/assets/320066/afa543d4-a886-4c4e-85a3-9383a9b74806)

2. Different paddings
![Screenshot 2023-06-26 at 4 05 54 PM](https://github.com/streamr-dev/core-frontend/assets/320066/22c7aa25-a7ed-4788-97a1-ba45e8730169)

For the seconds point, let's give all items the same vertical padding and to keep the visual space for edge items let's add some padding to the list itself.

pmichalowski commented on app/src/shared/components/Layout/Nav.styles.tsx:
```diff
@@ -0,0 +1,368 @@
+    highlight?: boolean
+    children: any
+}
+const UnstyledNavbarLink: FunctionComponent<UnstyledNavbarLinkProps> = ({
+    highlight,
+    children,
+    ...props
+}) => {
+    return <LinkWrapper {...props}>{children}</LinkWrapper>
+}
```
TBH it's not even a part of what I was doing - you see it as changes because I moved the styles to a separate file, but of course I will implement the suggested changes :)

pmichalowski replied:
also I do not actually get what you mean here and now when I look at it I think that best would be to remove this prop from the props type, it's being defined later by the "styled" component

mondoreale replied:
Yeah, it'd be totally fine (better?) to style `LinkWrapper` right away, directly.

And about naming props see https://styled-components.com/docs/api#transient-props.

pmichalowski commented:
Fixed the border radius and added padding to the list, but it doesn't look great either 
![Screenshot 2023-06-27 at 13 13 25](https://github.com/streamr-dev/core-frontend/assets/32245607/26539c0f-0405-4080-9dc5-d76fd966bca3)
